src/ModelMerger.py

```diff
@@ -0,0 +1,158 @@
+                    met_t.id = re.sub(r"[^_]+$", "t", met.id)  # substitutes the compartment suffix (all
+                                                                            # that follows the last underscore) with
+                                                                            # the new compartment's symbol
+                    met_t.compartment = "t"
+                    ex.add_metabolites({met_t: 1.0})
+                else:
+                    raise ValueError(ex.id + "should contain only one metabolite")
+            else:
+                raise ValueError("The exchange reaction " + ex.id + "should start with the 'EX_' prefix")
+
+    def translate_metabolites(self, to_translate, matches_df, score_type):
+        """
+        Translates a list of metabolites from namespace 2 to the best matching metabolite in namespace 1.
+
+        This function takes a list of metabolite IDs from namespace 2 and attempts to translate each
+        to the best matching metabolite ID in namespace 1 based on a matching score. The function ensures
+        that no two metabolites from the to_translate list are translated to the same ID in namespace 1.
+        If two or more metabolites have the same best match, only the one with the highest score
+        (or the first alphanumerically in case of a tie) is assigned. Unassigned metabolites will retain
+        their original IDs.
+
+        Parameters:
+            to_translate (list): A list of metabolites that need to be translated.
+            matches_df (DataFrame): A DataFrame containing the potential matches between namespaces,
+                                    with columns ['source_namespace', 'target_namespace', score_type].
+            score_type (string): A string indicating which column from matches_df contains the matching
+                                 score that will be used to rank the metabolite matches.
+        """
+        # # Sort the metabolites list alphanumerically by their ID
+        # to_translate = sorted(to_translate, key=lambda metabolite: metabolite.id)
+
+        # Remove "_t" suffix from the metabolite IDs
+        for met in to_translate:
+            met.id = re.sub(r"_t$", "", met.id)
+
+        sorted_matches = matches_df.sort_values(by=[score_type, 'source_namespace', 'target_namespace'],
+                                                ascending=[False, True, True])
+
+        best_matches = dict()
+
+        # Iterate over the to_translate list
+        for met_id in sorted_matches["source_namespace"]:
+            if met_id in [m.id for m in to_translate]:
+                #print([m.id for m in to_translate])
+                #print(met_id)
+                # Filter the matches for the current metabolite
+                matches_for_met = sorted_matches[sorted_matches['source_namespace'] == met_id]
+                # Assign the best available match
+                for _, row in matches_for_met.iterrows():
```

Comment:
By the sorting above matches_for_met will always contain the best match in the first row. So just pick the first row (Note: Is it possible that this df is empty?).

Reply:
Please refer in comment to the fact that this relies on the ordering established above